### DISCUSSION TITLE
Color-specific style overrides

### DISCUSSION THREAD

**Discussion by jerguslejko** [mantinedev/mantine#709] [Q&A] [answered]
We are using `<Button variant="light" color="gray" />` as a "secondary" button in our app. However, the color contrast on this button is a not great, so we're utilising [Styles on MantineProvider](https://mantine.dev/theming/mantine-provider/#styles-on-mantineprovider) to override the color with a darker gray like so:

```tsx
<MantineProvider
  styles={{
    Button: theme => ({
      light: { color: theme.colors.gray[8] },
    }),
  }}
>
  ...
</MantineProvider>
```

While this works for the gray button, it breaks all other `variant="light"` buttons. See [this codesandbox](https://codesandbox.io/s/romantic-hooks-x7m2v?file=/src/App.tsx) for how it breaks `variant="light" color="red"` buttons (I'd like the text to be red there).

Any suggestions on how to accomplish what I'm trying to do? I want to darken the gray on `variant="light" color="gray"` button without affecting the other styles.

Thanks

**Comment by rtivital** [comment, marked as answer]
Currently, it's not supported to get component params in provider styles, this option will be available in upcoming 4.0

**Reply by jerguslejko** [reply]
right. what do you recommend instead?

**Reply by rtivital** [reply]
Create a wrapper component that will check color and apply corresponding styles

**Reply by jerguslejko** [reply]
right, makes sense. 

is there a roadmap for v4?

**Reply by rtivital** [reply]
No we do not share a public roadmap

**Reply by jerguslejko** [reply]
understandable :) 

have a good one @rtivital
